admin/class-manage-fonts.php

```diff
@@ -53,6 +58,51 @@ function can_read_and_write_font_assets_directory () {
 	}
 
 
+    function manage_fonts_admin_page () {
+        // Load the required WordPress packages.
+        // Automatically load imported dependencies and assets version.
+        $asset_file = include plugin_dir_path( __DIR__ ) . 'build/index.asset.php';
+     
+        // Enqueue CSS dependencies.
+        foreach ( $asset_file['dependencies'] as $style ) {
+            wp_enqueue_style( $style );
+        }
+     
+        // Load our app.js.
+        wp_enqueue_script( 'create-block-theme-app', plugins_url( 'build/index.js', __DIR__ ), $asset_file['dependencies'], $asset_file['version'] );
+
+        wp_enqueue_style( 'manage-fonts-styles',  plugin_dir_url( __DIR__ ) . '/css/manage-fonts.css', array(), '1.0', false );
+
+        $theme_name = wp_get_theme()->get( 'Name' );
+
+        $theme_data = WP_Theme_JSON_Resolver::get_theme_data();
+        $theme_settings = $theme_data->get_settings();
+        $theme_font_families = $theme_settings['typography']['fontFamilies']['theme'];
+
+        if ( ! empty( $_POST['new-theme-fonts-json'] ) ) {
+            $theme_font_families = json_decode( stripslashes( $_POST['new-theme-fonts-json'] ), true );
+        }
+
+        $fonts_json = wp_json_encode( $theme_font_families, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE );
+        $fonts_json_string = preg_replace ( '~(?:^|\G)\h{4}~m', "\t", $fonts_json );
+        
+    ?>
+    <div class="wrap">
+        <h1 class="wp-heading-inline"><?php _e('Manage Theme Fonts', 'create-block-theme'); ?></h1>
+        <a href="<?php echo admin_url( 'themes.php?page=add-google-font-to-theme-json' ); ?>" class="page-title-action"><?php _e('Add Google Font', 'create-block-theme'); ?></a>
+        <a href="<?php echo admin_url( 'themes.php?page=add-local-font-to-theme-json' ); ?>" class="page-title-action"><?php _e('Add Local Font', 'create-block-theme'); ?></a>
+        <hr class="wp-header-end" />
+        <p><?php printf( esc_html__( 'These are the fonts currently available in your theme (%1$s).', 'create-block-theme' ), esc_html( $theme_name ) ); ?></p>
```

Comment:
If the theme itself bundles a custom font, should it display them? For example, I tested with Pixl which bundles DMMono, but I do not see it: 

<img width="1624" alt="Screen Shot 2022-10-18 at 12 32 57 PM" src="https://user-images.githubusercontent.com/5375500/196418416-cc575aa6-26ea-4e0e-aac3-b9bb0fd503d0.png">

Reply:
Yep, all the fonts present in theme.json should be listed. This was a glitch happening with some themes. It should be solved with the latest changes.

Reply:
I see it working now!